logsmanagement/helper.h

```diff
@@ -0,0 +1,330 @@
+#include <assert.h>
+#include <stdarg.h>
+#include <string.h>
+#include <sys/time.h>
+
+#define MAX_LOGS_MANAG_LOG_MSG_LEN 256
+
+typedef enum { LOGS_MANAG_ERROR,
+               LOGS_MANAG_WARNING,
+               LOGS_MANAG_INFO,
+               LOGS_MANAG_DEBUG } Log_level;
+
+#ifdef LOGS_MANAG_DEBUG_LEV
+#define fprintf_log(log_level, ...)                          \
+    do {                                                     \
+        if (1) fprintf_log_internal(log_level, __VA_ARGS__); \
+    } while (0)
+#else
+#define fprintf_log(log_level, ...)                          \
+    do {                                                     \
+        if (0) fprintf_log_internal(log_level, __VA_ARGS__); \
+    } while (0)
+#endif  // LOGS_MANAG_DEBUG_LEV
+
+#ifndef m_assert
+#if LOGS_MANAG_DEBUG_LEV                                             // Disable m_assert if production release
+#define m_assert(expr, msg) assert(((void)(msg), (expr))) /**< Custom assert function that prints out failure message */
+#else
+#define m_assert(expr, msg) \
+    do {                    \
+    } while (0)
+#endif  // LOGS_MANAG_DEBUG_LEV
+#endif  // m_assert
+
+// Portable thread local, see https://stackoverflow.com/questions/18298280/how-to-declare-a-variable-as-thread-local-portably
+#ifndef thread_local
+#if __STDC_VERSION__ >= 201112 && !defined __STDC_NO_THREADS__
+#define thread_local _Thread_local
+#elif   defined _WIN32 && ( \
+        defined _MSC_VER || \
+        defined __ICL || \
+        defined __DMC__ || \
+        defined __BORLANDC__ )
+#define thread_local __declspec(thread) 
+/* note that ICC (linux) and Clang are covered by __GNUC__ */
+#elif   defined __GNUC__ || \
+        defined __SUNPRO_C || \
+        defined __xlC__
+#define thread_local __thread
+#else
+#error "Cannot define thread_local"
+#endif
+#endif // thread_local
+
+#ifdef ENABLE_LOGSMANAGEMENT_TESTS
+#define UNIT_STATIC
+#else 
+#define UNIT_STATIC static
+#endif // ENABLE_LOGSMANAGEMENT_TESTS 
+
+#ifndef COMPILE_TIME_ASSERT // https://stackoverflow.com/questions/3385515/static-assert-in-c
+#define STATIC_ASSERT(COND,MSG) typedef char static_assertion_##MSG[(!!(COND))*2-1]
+// token pasting madness:
+#define COMPILE_TIME_ASSERT3(X,L) STATIC_ASSERT(X,static_assertion_at_line_##L)
+#define COMPILE_TIME_ASSERT2(X,L) COMPILE_TIME_ASSERT3(X,L)
+#define COMPILE_TIME_ASSERT(X)    COMPILE_TIME_ASSERT2(X,__LINE__)
+#endif  // COMPILE_TIME_ASSERT
+
+#define BIT_SET(a, b) ((a) |= (1ULL << (b)))
+#define BIT_CLEAR(a, b) ((a) &= ~(1ULL << (b)))
+#define BIT_CHECK(a, b) (!!((a) & (1ULL << (b))))  // '!!' to make sure this returns 0 or 1
+
+// Include for sleep_ms()
+#ifdef _WIN32
+#include <Windows.h>
+#else
+#include <unistd.h>
+#endif  // _WIN32
+
+extern uv_loop_t *main_loop; 
+
+/**
+ * @brief Get unix time in milliseconds
+ * @return Unix time in milliseconds
+ * @todo Replace gettimeofday() with uv_gettimeofday()
+ */
+static inline uint64_t get_unix_time_ms() {
```

Comment:
Please, take a look in [this](https://github.com/netdata/netdata/tree/master/libnetdata/clocks) library, you do not need to bring this function.